rust/src/detect/vlan_id.rs

```diff
@@ -0,0 +1,199 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::uint::{detect_parse_uint, DetectUintData};
+use std::ffi::CStr;
+use std::str::FromStr;
+
+pub const DETECT_VLAN_ID_ANY: i8 = i8::MIN;
+pub const DETECT_VLAN_ID_ALL: i8 = i8::MAX;
+pub const DETECT_VLAN_ID_COUNT: i8 = 100;
```

Comment:
We may want to transfer `src/decode-vlan.h:#define VLAN_MAX_LAYERS    3` here and use `VLAN_MAX_LAYERS` instead of hardcoded 3

Reply:
I defined ``VLAN_MAX_LAYERS`` in the ``vlan_id.rs`` file like this: ``pub const VLAN_MAX_LAYERS: i8 = 3;``
but I get this error message:

```
decode.h:503:22: error: ‘VLAN_MAX_LAYERS’ undeclared here (not in a function)
503 |     uint16_t vlan_id[VLAN_MAX_LAYERS];
```

And when I include ``rust.h`` in the ``decode.h`` file, this first error goes away, but I get several error messages similar to this one:

```
util-file.h:214:39: error: unknown type name ‘Flow’
214 | void FileDisableStoringForTransaction(Flow *f, const uint8_t direction, void *tx, uint64_t tx_id);
```

Reply:
I think you need to maintain 2 copies of the constant. One on the C side and the other one on the Rust side. The important thing is to avoid using magic numbers.

Reply:
Do not spend too much time on it if it gets too complex with include order.
Just do as Shivani said, and add some comments